File: JSBASIC/196_zeros_to_end.js
// 196	196_zeros_to_end.js	Move all zeros in an array to the end while keeping other order intact.

let arr = [1, 0, 3, 0, 5, 0, 7];
let nonZeros = arr.filter(x => x !== 0);
let zeros = arr.filter(x => x === 0);
let result = [...nonZeros, ...zeros];
console.log("Original:", arr);
console.log("Zeros moved to end:", result);

File: JavaBasic/196_zeros_to_end.java
// 196	196_zeros_to_end.java	Move all zeros in an array to the end while keeping other order intact.
import java.util.*;
public class _196_zeros_to_end {
    public static void main(String[] args) {
        int[] arr = {1,0,3,0,5,0,7};
        List<Integer> nonZero=new ArrayList<>(), zeros=new ArrayList<>();
        for (int x:arr) if(x!=0) nonZero.add(x); else zeros.add(x);
        nonZero.addAll(zeros);
        System.out.println("Zeros to end: " + nonZero);
    }
}
